autopush/router/fcm.py

```diff
@@ -32,8 +31,87 @@ def __init__(self, ap_settings, router_conf):
         self.auth = router_conf.get("auth")
         self.metrics = ap_settings.metrics
         self._base_tags = []
+        self.creds = {}
```

Comment:
I'd like to kill self.creds and self.regid because they get mutated during register/route calls. We can explicitly pass the creds/regid from router_data to _process_reply (or just pass the whole router_data dict itself), then the vars on self won't be necessary